server.py

```diff
@@ -1,28 +1,85 @@
-import socket
+import select, socket, sys, Queue
+
+inputs = []
+outputs = []
+message_queues = {}
+data_to_send = 0
+
+def add_new_client(new_socket):
+    connection, client_address = new_socket.accept()
+    connection.setblocking(0)
+    inputs.append(connection)
+    #message_queues[connection] = Queue.Queue()
+
+
+def handle_client_input(socket_to_read):
+    try:
+        data = socket_to_read.recv(1024)
+    except socket.error, e:
+        print "Client aborted " + str(e)
+
+        if socket_to_read in outputs:
+            outputs.remove(socket_to_read)
+
+        inputs.remove(socket_to_read)
+        socket_to_read.close()
+        #del message_queues[socket_to_read]
+        return False
+
+    if data:
+        global data_to_send
+        data_to_send = data + data
+        if socket_to_read not in outputs:
+            outputs.append(socket_to_read)
+    else:
+        if socket_to_read in outputs:
+            outputs.remove(socket_to_read)
+        inputs.remove(socket_to_read)
+        socket_to_read.close()
+        #del message_queues[socket_to_read]
+
+    return True
+
+
+def handle_client_output(socket_to_write):
+    global data_to_send
```

Comment:
Please check how the code behaves when you have multiple clients connected

Modify one of your clients so instead of taking an input, it always sends "Hello + random number", and waits 1 second between each loop using `time.sleep(1)`, then try decreasing the time to 0.2 or lower and see if anything strange happens